test/unit/index.test.ts

```diff
@@ -0,0 +1,161 @@
+  ['GridFSBucket',                       'reason'],
+  ['ListCollectionsCursor',              'reason'],
+  ['ListIndexesCursor',                  'reason'],
+  ['Logger',                             'reason'],
+  ['MongoClient',                        'reason'],
+  // global promise setter
+  ['Promise',                            'global promise setter'],
+  // enums
+  ['BatchType',                          'reason'],
+  ['GSSAPICanonicalizationValue',        'reason'],
+  ['AuthMechanism',                      'reason'],
+  ['Compressor',                         'reason'],
+  ['CURSOR_FLAGS',                       'reason'],
+  ['AutoEncryptionLoggerLevel',          'reason'],
+  ['MongoErrorLabel',                    'reason'],
+  ['ExplainVerbosity',                   'reason'],
+  ['LoggerLevel',                        'reason'],
+  ['ServerApiVersion',                   'reason'],
+  ['BSONType',                           'reason'],
+  ['ReturnDocument',                     'reason'],
+  ['ProfilingLevel',                     'reason'],
+  ['ReadConcernLevel',                   'reason'],
+  ['ReadPreferenceMode',                 'reason'],
+  ['ServerType',                         'reason'],
+  ['TopologyType',                       'reason'],
+  ['ReadConcern',                        'reason'],
+  ['ReadPreference',                     'reason'],
+  ['WriteConcern',                       'reason'],
+  // events
+  ['CommandFailedEvent',                 'reason'],
+  ['CommandStartedEvent',                'reason'],
+  ['CommandSucceededEvent',              'reason'],
+  ['ConnectionCheckedInEvent',           'reason'],
+  ['ConnectionCheckedOutEvent',          'reason'],
+  ['ConnectionCheckOutFailedEvent',      'reason'],
+  ['ConnectionCheckOutStartedEvent',     'reason'],
+  ['ConnectionClosedEvent',              'reason'],
+  ['ConnectionCreatedEvent',             'reason'],
+  ['ConnectionPoolClearedEvent',         'reason'],
+  ['ConnectionPoolClosedEvent',          'reason'],
+  ['ConnectionPoolCreatedEvent',         'reason'],
+  ['ConnectionPoolMonitoringEvent',      'reason'],
+  ['ConnectionReadyEvent',               'reason'],
+  ['ServerClosedEvent',                  'reason'],
+  ['ServerDescriptionChangedEvent',      'reason'],
+  ['ServerHeartbeatFailedEvent',         'reason'],
+  ['ServerHeartbeatStartedEvent',        'reason'],
+  ['ServerHeartbeatSucceededEvent',      'reason'],
+  ['ServerOpeningEvent',                 'reason'],
+  ['TopologyClosedEvent',                'reason'],
+  ['TopologyDescriptionChangedEvent',    'reason'],
+  ['TopologyOpeningEvent',               'reason'],
+  ['SrvPollingEvent',                    'reason'],
+
+  ['GridFSBucketReadStream',             'reason'],
+  ['GridFSBucketWriteStream',            'reason'],
+  ['OrderedBulkOperation',               'reason'],
+  ['UnorderedBulkOperation',             'reason'],
+
+  // TS-NODE Adds these keys but they are undefined... weird
```

Comment:
do we want to do any further investigation into this? having a comment saying that this is weird is a bit of an anti-pattern for the repo - it's not going to be anymore clear to the next person coming across this

Reply:
Yea I have no idea what makes these two props special, they don't exist when I import from lib, only when I run the driver in ts-node do they appear. I think its fine to live with, I can clarify the comment. The assertions that these remain equal to undefined should cover us.

I don't want to import from lib because it will block the whole mocha run if we push a small accidental TS issue (annoying when someone is iterating a lot)